skyportal/handlers/api/internal/profile.py

```diff
@@ -148,96 +148,103 @@ def patch(self, user_id=None):
                 schema: Error
         """
         data = self.get_json()
-        if user_id is None:
-            user = User.get_if_accessible_by(
-                self.associated_user_object.id, self.current_user, mode="update"
-            )
-        else:
-            user = User.get_if_accessible_by(user_id, self.current_user, mode="update")
 
-        if data.get("username") is not None:
-            username = data.pop("username").strip()
-            if username == "":
-                return self.error("Invalid username.")
-            if len(username) < 5:
-                return self.error("Username must be at least five characters long.")
-            user.username = username
+        with self.Session() as session:
+            if user_id is None:
+                user_id = self.associated_user_object.id
+            user = session.scalars(
+                User.select(session.user_or_token, mode="update").where(
+                    User.id == user_id
+                )
+            ).first()
+            if user is None:
+                return self.error(f'Cannot find User with ID: {user_id}')
 
-        if data.get("first_name") is not None:
-            user.first_name = data.pop("first_name")
+            if data.get("username") is not None:
+                username = data.pop("username").strip()
+                if username == "":
+                    return self.error("Invalid username.")
```

Comment:
We don't actually need to check empty string if checking length.

Reply:
good point.